### ISSUE TITLE
Snapshots (sort of a cheap undo)

### ISSUE THREAD

**Issue by baconpaul** [surge-synthesizer/surge#1217]
General undo is hard - some daws provide undo of exposed params but lots of things like the wt and the modulation are not exposed. And building an undo stack in the ui is a lot of work

Evil and I were discussing and we thought a halfway point may be to snapshot every 60 seconds, take 10 snapshots, never snapshot if transport is running, expose it all as a user pref, make sure snapshots for multiple instances don’t collide. That kinda thing could be useful. Sort of an auto save which runs when your daw is not playing. Maybe do some rudimentary check of if anything has changed (perhaps hash the streamed patch?)

Not gonna do it now but wanted to capture the convo in the list

**Comment by mkruselj** [comment]
No actually I didn't mean like that :) I meant like physical snapshots - i.e. you get 10 slots per patch so you can save various changes along the way. In essence variations of the patch. Example: See GForce Minimonsta. You get the "Melohman octave" in which you can save up to 12 variations of a patch and keyswitch between them. Can be useful both as a live performance tool, but also as patch development tool.

**Comment by baconpaul** [comment]
Yeah alchemy does exactky the same thing 

It’s a very useful feature.

We can use one slot as the “time machine” and combine both ideas

**Comment by itsmedavep** [comment]
Unless I misunderstand, that will only help in the cases that you remember to create a snapshot.

All other cases where you accidentally change something without saving a "snapshot" wont be helped. 

I thought the initial concern from Slack was that it is easy to accidentally loose changes to your patch. Thus the desire for an undo.  

Snapshots are good. Just solving a differing problem IMO.

**Comment by mkruselj** [comment]
@baconpaul Why close this perfectly valid FR?

**Comment by baconpaul** [comment]
I'm gonna close this because we are just gonna do proper undo, OK?

**Comment by baconpaul** [comment]
Yeah that’s a very different fr and probably. Ore Kim a scene morph or modulation supermacro
